### PR TITLE
feat: support UnixFSv1.5 metadata

### PR DESCRIPTION
**Pull request by achingbrain**
Depends on:

- [ ] https://github.com/ipfs/js-ipfs-http-client/pull/1186

### CODE REVIEW

File: src/parser.js [hunk 1]
```diff
@@ -50,49 +44,90 @@ const ignore = async (stream) => {
...
+
+    if (!part.headers['content-disposition']) {
+      throw new Error('No content disposition in multipart part')
+    }
 
-      await ignore(part.body)
+    const disposition = parseDisposition(part.headers['content-disposition'])
 
-      continue
+    if (disposition.name.includes('mtime')) {
```
Comment (alanshaw) [inline src/parser.js]: Ok so, metadata must preceed the file data if set. Can you document the proposal so we can check this is ok with the go-ipfs team?

File: src/parser.js [hunk 2]
```diff
@@ -50,49 +44,90 @@ const ignore = async (stream) => {
...
+      entry.mtime = parseInt((await collect(part.body)).toString('utf8'), 10)
     }
 
-    if (partHeader.mime === applicationSymlink) {
-      const target = await collect(part.body)
+    if (disposition.name.includes('mode')) {
```
Comment (alanshaw) [inline src/parser.js]: Is there a reason for `includes` rather than `disposition.name === 'mode'`?

File: src/parser.js (hunk 1 above)
Comment (achingbrain) [reply src/parser.js]: This has been changed to be passed as a header instead of a multipart part

File: src/parser.js (hunk 2 above)
Comment (achingbrain) [reply src/parser.js]: This has been refactored out

**Comment by achingbrain** [comment]
Superseded by #39

**Pull request closed without merging**